src/groups/mqb/mqbblp/mqbblp_clusterqueuehelper.cpp

```diff
@@ -2080,6 +2080,13 @@ bsl::shared_ptr<mqbi::Queue> ClusterQueueHelper::createQueueFactory(
     // Domain is already aware of the queue, reuse it.
     bsl::shared_ptr<mqbi::Queue> iQueueSp;
     if (context.d_domain_p->lookupQueue(&iQueueSp, queueContext->uri()) == 0) {
+        /// In rare situations we call `resetButKeepPending` and
+        /// `d_liveQInfo->d_queue_sp` is reset, but the corresponding domain
+        /// still contains a shared_ptr to this queue.
+        /// We must restore `d_queue_sp` from the domain.
+        queueContext->d_liveQInfo.d_queue_sp = bsl::shared_ptr<mqbblp::Queue>(
+            iQueueSp,
+            static_cast<mqbblp::Queue*>(iQueueSp.get()));
```

Comment:
This is the preferred way of `shared_ptr` casting from BDE docs
https://bloomberg.github.io/bde-resources/doxygen/bde_api_prod/group__bslstl__sharedptr.html

```
Explicit Casting
Through "aliasing", a shared pointer of any type can be explicitly cast to a shared pointer of any other type using any legal cast expression. For example, to statically cast a shared pointer to type A (shared_ptr<A>) to a shared pointer to type B (shared_ptr<B>), one can simply do the following:

shared_ptr<A> spa;
shared_ptr<B> spb(spa, static_cast<B *>(spa.get()));
```